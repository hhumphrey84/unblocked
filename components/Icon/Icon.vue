<template>
    <svg
        :aria-labelledby="id"
        :data-test-id="dataTestId"
        viewBox="0 0 60 60"
        xmlns="http://www.w3.org/2000/svg"
    >
        <title :id="id">{{ title }}</title>
        <slot />
    </svg>
</template>

<script>
    import testIds from './testIds.constants';

    const props = {
        dataTestId: {
            type: String,
            required: false,
            default: testIds.WRAPPER,
        },
        title: {
            type: String,
            required: true,
        },
        id: {
            type: String,
            required: true,
        },
    }

    export default {
        props,
    };
</script>
