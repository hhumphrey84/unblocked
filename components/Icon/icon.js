const props = {
    title: {
        type: String,
        required: true
    },
    id: {
        type: String,
        required: true
    },
}

export default {
    props,
};
