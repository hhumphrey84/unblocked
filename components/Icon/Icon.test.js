import { shallowMount } from '@vue/test-utils';
import Icon from './Icon.vue';

describe('Icon', () => {
    test('mounts properly', () => {
        const wrapper = shallowMount(Icon, {
            propsData: {
                title: 'SVG Title',
                id: 'unique-id',
            },
            slots: {
                default: ['<path data-test-id="Icon-slot" />'],
            },
        })
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    test('renders properly', () => {
        const wrapper = shallowMount(Icon, {
            propsData: {
                title: 'SVG Title',
                id: 'unique-id',
            },
            slots: {
                default: ['<path data-test-id="Icon-slot" />'],
            },
        })
        expect(wrapper.html()).toMatchSnapshot()
    })
})
