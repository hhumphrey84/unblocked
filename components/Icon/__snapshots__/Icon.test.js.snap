// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders properly 1`] = `
<svg aria-labelledby="unique-id" viewBox="0 0 20 21" xmlns="http://www.w3.org/2000/svg">
  <title id="unique-id">SVG Title</title>
  <path data-test-id="Icon-slot"></path>
</svg>
`;
